### ISSUE TITLE
504 Lambda response code when updating

### ISSUE THREAD

**Issue by ThibaultMetais** [Miserlou/Zappa#1590]
Hello ! 

I am actually trying to deploy a Django application on AWS Lambda and I got this issue when I try to update

```
Calling update for stage dev..
Downloading and installing dependencies..
 - scipy==1.1.0: Using locally cached manylinux wheel
 - psycopg2==2.7.5: Using locally cached manylinux wheel
 - psycopg2-binary==2.7.5: Using locally cached manylinux wheel
 - pandas==0.23.4: Using locally cached manylinux wheel
 - numpy==1.15.0: Using locally cached manylinux wheel
 - sqlite==python36: Using precompiled lambda package
Packaging project as gzipped tarball.
Target directory /Users/ravisha/Documents/MyFleetBuy_Server/myfleetbuy/handler_venv/lib/python3.6/site-packages/zappa-0.46.2.dist-info already exists. Specify --upgrade to force replacement.
Target directory /Users/ravisha/Documents/MyFleetBuy_Server/myfleetbuy/handler_venv/lib/python3.6/site-packages/zappa already exists. Specify --upgrade to force replacement.
You are using pip version 10.0.1, however version 18.0 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Downloading and installing dependencies..
 - sqlite==python36: Using precompiled lambda package
Packaging project as zip.
Uploading myfleetbuy-dev-1534529408.tar.gz (151.5MiB)..
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 159M/159M [00:25<00:00, 6.12MB/s]
Uploading handler_myfleetbuy-dev-1534529655.zip (10.5MiB)..
100%|███████████████████████████████████████████████████████████████████████████████████████████| 11.0M/11.0M [00:01<00:00, 5.52MB/s]
Updating Lambda function code..
Updating Lambda function configuration..
Uploading myfleetbuy-dev-template-1534529692.json (1.6KiB)..
100%|███████████████████████████████████████████████████████████████████████████████████████████| 1.62K/1.62K [00:00<00:00, 32.3KB/s]
Deploying API Gateway..
Scheduling..
Unscheduled myfleetbuy-dev-zappa-keep-warm-handler.keep_warm_callback.
Scheduled myfleetbuy-dev-zappa-keep-warm-handler.keep_warm_callback with expression rate(4 minutes)!
Error: Warning! Status check on the deployed lambda failed. A GET request to '/' yielded a 504 response code.
```

Also zappa tail seems to display this error but I don't know if it's linked 

```
[1534520729301] Instancing..
[1534520729303] Failed to find library...right filename?
[1534520729337] No module named 'django': ModuleNotFoundError
Traceback (most recent call last):
  File "/var/task/handler.py", line 567, in lambda_handler
  return LambdaHandler.lambda_handler(event, context)
  File "/var/task/handler.py", line 237, in lambda_handler
  handler = cls()
  File "/var/task/handler.py", line 143, in __init__
  wsgi_app_function = get_django_wsgi(self.settings.DJANGO_SETTINGS)
  File "/var/task/zappa/ext/django_zappa.py", line 9, in get_django_wsgi
  from django.core.wsgi import get_wsgi_application
ModuleNotFoundError: No module named 'django'
```

I am on MacOS and my pip freeze results are 

```
argcomplete==1.9.3
base58==1.0.0
boto3==1.7.78
botocore==1.10.78
certifi==2018.8.13
cfn-flip==1.0.3
chardet==3.0.4
click==6.7
Django==2.1
django-filter==2.0.0
django-oauth-toolkit==1.2.0
django-rest-framework==0.1.0
django-webpack-loader==0.6.0
djangorestframework==3.8.2
docutils==0.14
durationpy==0.5
future==0.16.0
hjson==3.0.1
idna==2.7
jmespath==0.9.3
kappa==0.6.0
lambda-packages==0.20.0
numpy==1.15.0
oauthlib==2.1.0
pandas==0.23.4
placebo==0.8.1
psycopg2==2.7.5
psycopg2-binary==2.7.5
python-dateutil==2.6.1
python-slugify==1.2.4
pytz==2018.5
PyYAML==3.12
requests==2.19.1
s3transfer==0.1.13
scipy==1.1.0
six==1.11.0
toml==0.9.4
tqdm==4.19.1
troposphere==2.3.1
Unidecode==1.0.22
urllib3==1.23
Werkzeug==0.14.1
wsgi-request-logger==0.4.6
zappa==0.46.2

```
 And my zappa settings

```
{
    "dev": {
        "aws_region": "us-west-1",
        "django_settings": "myfleetbuy.settings",
        "profile_name": "default",
        "project_name": "myfleetbuy",
        "runtime": "python3.6",
        "s3_bucket": "zappa-3uk7ut9lh",
        "slim_handler": "true"
    }
}
```

Thanks !

**Comment by trumpet2012** [comment]
@ThibaultMetais Did you figure out what was causing this? I'm currently experiencing the same problem.

**Comment by ThibaultMetais** [comment]
> @ThibaultMetais Did you figure out what was causing this? I'm currently experiencing the same problem.

Hi ! Unfortunately I couldn't find a solution : I was forced to put  "slim_handler": "true" because of the size of the project, but it seems like this option is causing a lot of problems...

Anyway, I personally think that Lambda is meant to realize simple tasks or host simple and light web apps. Then, I strongly recommend ElasticBeanstalks which is way more easy to set up and maintain, but also way more adapted to this kind of projects. But once again, it's only from my experience.

Hope that helps a little bit, good luck with that !

**Comment by 3nomis** [comment]
Same proble. did you find a solution?

**Comment by rolexsanches** [comment]
same problem

**Comment by trumpet2012** [comment]
@3nomis @rolexsanches I'm sorry but I don't remember exactly what fixed this for us. We did however reduce the size of our deployed code which I believe helped a lot of the problems we were having.

Just in case it helps, as part of our deployment we remove nearly all of the code in our `node_modules/` after we run webpack. This helped save a lot of space. We also compiled our own versions of some of our python dependencies such as opencv, numpy and lxml. We were able to compile just the parts that we needed in order to save space.
